src/biome.rs

```diff
@@ -1,31 +1,53 @@
-use std::{env, fs};
-use zed_extension_api::{self as zed, LanguageServerId, Result};
+use std::{env, fs, path::Path};
+use zed_extension_api::{self as zed, serde_json, LanguageServerId, Result};
 
-const SERVER_PATH: &str = "node_modules/.bin/biome";
+const SERVER_PATH: &str = "node_modules/@biomejs/biome/bin/biome";
 const PACKAGE_NAME: &str = "@biomejs/biome";
 
-struct BiomeExtension {
-  did_find_server: bool,
-}
+struct BiomeExtension;
 
 impl BiomeExtension {
-  fn server_exists(&self) -> bool {
-    fs::metadata(SERVER_PATH).map_or(false, |stat| stat.is_file())
+  fn server_exists(&self, path: &str) -> bool {
+    fs::metadata(path).map_or(false, |stat| stat.is_file())
   }
 
-  fn server_script_path(&mut self, language_server_id: &LanguageServerId) -> Result<String> {
-    let server_exists = self.server_exists();
-    if self.did_find_server && server_exists {
-      return Ok(SERVER_PATH.to_string());
+  fn server_script_path(
+    &mut self,
+    language_server_id: &LanguageServerId,
+    worktree: &zed::Worktree,
+  ) -> Result<String> {
+    // This is a workaround, as reading the file from wasm doesn't work.
+    // Instead we try to read the `package.json`, see if `@biomejs/biome` is installed
+    let package_json = worktree
+      .read_text_file("package.json")
```

Comment:
Check the existence of the LSP binary in the workspace with `fs::metadata` doesn't work, probably because of wasm. Instead, we can read the `package.json` with `worktree::read_text_file` and find out if the user has installed the `@biomejs/biome`.

Originally I was trying to read the `node_modules/@biomejs/biome/package.json` with `worktree::read_text_file`, just to check if the user already installed the package, but it doesn't work for some reason.